plik ze zmianami

AnnaSkupinska opened the pull request:
Oto nowy plik, który ma wiele zmian.

Zauważyłam, że sam wprowadziłeś zmiany do projektu. Nie omówiłam ich z tobą, dlatego tutaj wprowadzam plik, który zawiera zmiany zrobione przeze mnie od tego czasu.

Główny problem z plikiem w master branch polega na tym, że nie jest w stanie zapamiętywać poprzednich ruchów, i wrócić się gdy potrzeba. Na potkałam ten problem parę razy, gdy kazałam algorytmowi rozwiązać rozwiązywalne sudoku. Wracanie się sprawia, że algorytm napotka na ścianę (pole któremu nie można przydzielić żadnego koloru) wraca się i próbuje inne kolory na poprzednich polach.

Trudno mi powiedzieć, czy heurystyka z http://www.nmk.fpv.ukf.sk/2014/proceedings/6_chytry_fulltext.pdf jest w stanie pokolorować graf bez wracania. Wiem natomiast, że mój algorytm działa nieco inaczej. VLASTIMIL CHYTRÝ tworzy graf w miarę kolorowania, podczas gdy moja wersja najpierw tworzy graf, a potem go koloruje.

AnnaSkupinska committed:
plik ze zmianami

Oto nowy plik, który ma wiele zmian.

Zauważyłam, że sam wprowadziłeś zmiany do projektu. Nie omówiłam ich z tobą, dlatego tutaj wprowadzam plik, który zawiera zmiany zrobione przeze mnie od tego czasu.

Główny problem z plikiem w master branch polega na tym, że nie jest w stanie zapamiętywać poprzednich ruchów, i wrócić się gdy potrzeba. Na potkałam ten problem parę razy, gdy kazałam algorytmowi rozwiązać rozwiązywalne sudoku. Wracanie się sprawia, że algorytm napotka na ścianę (pole któremu nie można przydzielić żadnego koloru) wraca się i próbuje inne kolory na poprzednich polach.

Trudno mi powiedzieć, czy heurystyka z http://www.nmk.fpv.ukf.sk/2014/proceedings/6_chytry_fulltext.pdf jest w stanie pokolorować graf bez wracania. Wiem natomiast, że mój algorytm działa nieco inaczej. VLASTIMIL CHYTRÝ tworzy graf w miarę kolorowania, podczas gdy moja wersja najpierw tworzy graf, a potem go koloruje.

PPokorski commented on matrix.py:
```diff
@@ -63,11 +72,9 @@ def colorGraph(self):
+			saturationDictionary[i] = self.getSaturation(i)
 			if self.colors[i] == 0:
 				setOfUncoloredPoints.add(i)
-				saturationDictionary[i] = self.getSaturation(i)
```
W #1 dodałaś tę opcję, usuwając to, co jest kilka linijek wyżej. Które jest aktualne?

AnnaSkupinska replied:
Hmm... to dotyczyło błędu na który się napotkałam. Wyglądało na to, że miało to jakiś związek z nieprawidłowo policzoną saturacją. Więc uznałam że lepiej updateować saturację dla wszystkich punktów, nie tylko tych niezamalowanych. A to dlatego że zamalowane punkty mogą zostać odmalowane.

AnnaSkupinska replied:
Poza tym, funkcja 'colorGraph' to stara wersja 'colorGraphNew'. Nie działa poprawnie bez nawracania się.

Powinnam ją skasować.

PPokorski commented on matrix.py:
```diff
@@ -93,11 +100,9 @@ def colorGraphOneStep(self):
+			saturationDictionary[i] = self.getSaturation(i)
 			if self.colors[i] == 0:
 				setOfUncoloredPoints.add(i)
-				saturationDictionary[i] = self.getSaturation(i)
```
J.w.

AnnaSkupinska replied:
Jak wyżej. To stara wersja.

AnnaSkupinska committed:
Więcej zmian

OK, usunęłam variable annotations, aby działao na pythonie 3.5.

Zmieniłam iterface, działa na linuxie

Wiem też, dlaczego mój algorytm się... "zapętluje". Próbowałam go zapętlić na mniejszych sudoku 4x4, ale nigdy mi się to nie udało. Potem, analizując kod odkryłam, że mój algorytm wcale się nie zapętluje. On po prostu rozróżnia dwie te same tablice pokolorowane w różnej kolejności jako różne, nawet jeżeli były pokolorowane tak samo. Więc... w tablicy 9x9 komputer musi się przekopać co najwyżej przez 80! możliwości zanim może stwierdzić że nie da się rozwiązać sudoku. Na tablicach 4x4 udawało mu się to, bo jest dużo mniej możliwości.

Muszę pomyśleć, jak rozwiązać ten problem

PPokorski commented on matrix.py:
```diff
@@ -35,39 +38,43 @@ def printColors(self):
+	def getLowestPossibleColorFromSaturationSet(self,sSet):
+		min = 10
+		for color in sSet:
+			if min > color:
```
A może po prostu https://docs.python.org/2/library/functions.html#min :)?

AnnaSkupinska replied:
Ahm... też może być :)

PPokorski commented on matrix.py:
```diff
@@ -1,29 +1,32 @@
-
+import math #dla operacji matematycznych
+import random	# dla generowania losowych plansz
+from msvcrt import getch	#dla interface z użytkownikiem (nieużywane)
```
Jeśli nieużywane to w sumie można skasować :)

AnnaSkupinska replied:
A w sumie prawda :)

PPokorski reviewed (changes requested):
Generalnie spoko :) Jak się odniesiesz do tych sugestii to można mergować :)
